### PR TITLE
refactor(CL): remove NumPools query from gamm

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #3240

## What is the purpose of the change

Since we are planning to merge gamm refactor to `main` soon, it might be best to apply this change prior to that so that the extra complexity stemming from `NumPools` query is not present in `main`.

Context: https://github.com/osmosis-labs/osmosis/issues/3240

### CODE REVIEW

**Comment by czarcas7ic** [comment]
It appears the 'osmosisd tx swaprouter create-pool' command is failing in e2e

File: x/gamm/keeper/genesis_test.go [hunk 1]
```diff
@@ -111,7 +109,6 @@ func TestGammExportGenesis(t *testing.T) {
 
 	genesis := app.GAMMKeeper.ExportGenesis(ctx)
 	require.Len(t, genesis.Pools, 2)
-	require.Equal(t, genesis.PoolCount, uint64(2))
```
Comment (czarcas7ic) [inline x/gamm/keeper/genesis_test.go:114]: shouldn't we be checking NextPoolID here?

File: x/gamm/simulation/sim_msgs.go [hunk 2]
```diff
@@ -297,6 +300,10 @@ func getRandPool(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Context) (uint64
 	poolCoins = simtypes.RemoveIndex(poolCoins, index)
 	coinOut := poolCoins[0]
 	poolDenoms := osmoutils.CoinsDenoms(pool.GetTotalPoolLiquidity(ctx))
-	gammDenom := types.GetPoolShareDenom(pool_id)
-	return pool_id, pool, coinIn, coinOut, poolDenoms, gammDenom, err
+
+	// pool ids start from 1.
+	poolId := poolIndex + 1
```
Comment (czarcas7ic) [inline x/gamm/simulation/sim_msgs.go:305]: if we are max bound by the upcoming pool ID, its possible the poolIndex is the current largest poolID, so if we add one to here wouldn't it be a non existent pool?

File: app/upgrades/v13/upgrades.go [hunk 3]
```diff
@@ -98,7 +98,6 @@ func migrateNextPoolId(ctx sdk.Context, gammKeeper *gammkeeper.Keeper, swaproute
 	// N.B: pool id in gamm is to be deprecated in the future
 	// Instead,it is moved to swaprouter.
 	nextPoolId := gammKeeper.GetNextPoolId(ctx)
-	gammKeeper.SetPoolCount(ctx, nextPoolId-1)
 	swaprouterKeeper.SetNextPoolId(ctx, nextPoolId)
```
Comment (czarcas7ic) [inline app/upgrades/v13/upgrades.go:101]: Without going through the very fun e2e debugging process yet, my first reaction to the e2e error is the nextPoolID is not getting properly set

**Review by czarcas7ic — COMMENTED** [review]
Leaving comments for now while we look into e2e failure

File: app/upgrades/v13/upgrades.go (hunk 3 above)
Comment (p0mvn) [reply app/upgrades/v13/upgrades.go:101]: @czarcas7ic hoping this: [0cf44a3](https://github.com/osmosis-labs/osmosis/pull/3492/commits/0cf44a3cef30d26bdc72da8bc1e3d5467c29ce20) fixed e2e. It was calling the removed query

File: x/gamm/keeper/genesis_test.go (hunk 1 above)
Comment (p0mvn) [reply x/gamm/keeper/genesis_test.go:114]: Next pool id is tracked in the swap router now. All increment logic was removed from gamm. The reason we still keep the index is for migrations

File: x/gamm/simulation/sim_msgs.go (hunk 2 above)
Comment (p0mvn) [reply x/gamm/simulation/sim_msgs.go:305]: I think this is right

Assume we have 3 pools with ids: 1, 2, 3.

So, the max `poolIndex` we can generate is 2.

Then, max `poolId` is 3 which is the correct one

**Comment by p0mvn** [comment]
If e2e passes, this is g2g since all other comments are addressed

**Comment by p0mvn** [comment]
@czarcas7ic this is ready

File: x/gamm/simulation/sim_msgs.go (hunk 2 above)
Comment (czarcas7ic) [reply x/gamm/simulation/sim_msgs.go:305]: yeah I think you are correct, the verbiage in the comment "max bound by next poolID" was confusing which is why I had the initial impression

**Review by czarcas7ic — APPROVED** [review]

**Pull request merged**